// account/index.js
/*
    methods {
        userExists(session)
        newUser(session)
        getUser(session)
        updateRefreshToken(session)
    }
*/

var fs = require('fs')
var path = require('path')

var filePath = path.join(__dirname, "../..", 'userData.json')

function readUsers() {
  if (!fs.existsSync(filePath)) 
    return []
  var data = fs.readFileSync(filePath);
  return JSON.parse(data);
}

function writeUsers(users) {
  fs.writeFileSync(filePath, JSON.stringify(users, null, 2));
}



module.exports = {
  userExists: function(session) {
    var users = readUsers()
    return users.some(function(user) {
        return user.id === session.user_id;
    })
  },

  newUser: function(session) {
    var users = readUsers();
    users.push({ 
        "id": session.user_id, 
        "refresh_token": session.refresh_token,
        "username": session.username,
        "pfp": session.pfp
    })
    writeUsers(users);
  },

  getUser: function(session) {
    var users = readUsers()
    var user =  users.find(function(user) {
        return user.id === session.user_id;
    });
    return user || null;
  },

  updateRefreshToken: function(session) {
    var users = readUsers()
    for (var i = 0; i < users.length; i++) {
      if (users[i].id === session.user_id) {
        users[i].refresh_token = session.refresh_token;
        break;
      }
    }
    writeUsers(users);
  }
};
